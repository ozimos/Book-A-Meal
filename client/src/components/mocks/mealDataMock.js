export const rows = [
  { id: 'abc', price: 1200 }, { id: 'def', price: 2000 }
];

export const allMeals = {
  data: { rows }

};

export const meal = {
  data: { id: 'mealId',
    price: 200,
    description: 'nice',
    imageUrl: 'nice',
    title: 'nice'
  }
};
