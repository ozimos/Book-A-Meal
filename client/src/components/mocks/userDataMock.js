export const userData = {
  data: { id: '123', email: 'asdf' },
  message: 'abc',
  token: 'abc'
};
export const userInput = {
  data: { email: 'asdf' },

};