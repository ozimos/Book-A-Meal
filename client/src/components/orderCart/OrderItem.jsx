import React from 'react';
import PropTypes from 'prop-types';
import format from 'date-fns/format';
import differenceInMinutes from 'date-fns/difference_in_minutes';
import isToday from 'date-fns/is_today';
import '../../../public/styles/order_item.scss';

const OrderItem = ({ id, Meals, addOrder, position, updatedAt }) => {
  const details = Meals.map(meal => (
    <div key={meal.id} className="row">
      <div className="col s7 text_left long_string left_pad">{meal.title}</div>
      <div className="col s3 text_left left_pad">{meal.price}</div>
      <div className="col s2 text_left">{meal.MealOrders.quantity}</div>
    </div>));
  const hasEditMinutes = (parseInt(process.env.ORDER_EDIT_MINUTES, 10) || 15)
       - differenceInMinutes(new Date(), updatedAt) > 0;

  return (
    <div className="row order_item" style={{ borderBottom: '1px solid' }}>
      <div className="col s1 text_left">{position}</div>
      <div className="col s9">
        {details}
      </div>
      <div className="col s1 text_left">{format(updatedAt, 'h:m A')}</div>
      <div className="col s1 text_right">
        {
          isToday(updatedAt) && hasEditMinutes &&
          <button onClick={() => addOrder(id)} className="btn title-button">Edit</button>
        }
      </div>
    </div>
  );
};

OrderItem.propTypes = {
  id: PropTypes.string.isRequired,
  updatedAt: PropTypes.string.isRequired,
  position: PropTypes.number.isRequired,
  Meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  addOrder: PropTypes.func.isRequired
};
export default OrderItem;

