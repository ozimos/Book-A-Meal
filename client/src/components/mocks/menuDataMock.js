import { allMeals } from './mealDataMock';

export default {
  data: { Meals: allMeals.data }
};