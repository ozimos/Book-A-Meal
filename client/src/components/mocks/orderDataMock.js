import { allMeals } from './mealDataMock';

const Meals = allMeals.data.map(meal => ({ ...meal, MealOrders: { quantity: 2 } }));
export const order = {
  data: { id: 'orderId',
    userId: 'userId',
    Meals,
    mealList: ['abc', 'def'],
    quantityList: [2, 2] }
};
export const allOrders = {
  data: [{ ...order.data, id: 'order1' }, { ...order.data, id: 'order2' }]
};