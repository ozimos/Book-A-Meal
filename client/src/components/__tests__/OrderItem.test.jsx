/* global React:false, shallow:false toJson:false */

import OrderItem from '../orderCart/OrderItem';
import { order } from '../mocks/orderDataMock';

describe('OrderItem Component', () => {

  it('should render without throwing an error', () => {
    expect(shallow(<OrderItem
      id={order.data.id}
      Meals={order.data.Meals}
      addOrder={jest.fn()}
    />)
      .exists(<div
        className="row"
        style={{ borderBottom: '1px solid' }}
      />)).toBe(true);
  });
  it('renders correctly', () => {
    const wrapper = shallow(<OrderItem
      id={order.data.id}
      Meals={order.data.Meals}
      addOrder={jest.fn()}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

