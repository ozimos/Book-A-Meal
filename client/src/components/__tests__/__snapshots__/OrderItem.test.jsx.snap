// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderItem Component renders correctly 1`] = `
<div
  className="row"
  style={
    Object {
      "borderBottom": "1px solid",
    }
  }
>
  <div
    className="col-4"
  >
    orderId
  </div>
  <div
    className="col-7"
  >
    <div
      className="row"
      key="abc"
    >
      <div
        className="col-8"
      />
      <div
        className="col-3"
      >
        1200
      </div>
      <div
        className="col-1"
      >
        2
      </div>
    </div>
    <div
      className="row"
      key="def"
    >
      <div
        className="col-8"
      />
      <div
        className="col-3"
      >
        2000
      </div>
      <div
        className="col-1"
      >
        2
      </div>
    </div>
  </div>
  <div
    className="col-1"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Edit
    </button>
  </div>
</div>
`;
